staging/src/k8s.io/pod-security-admission/admission/admission.go

```diff
@@ -603,18 +615,23 @@ func invalidResponse(attrs api.Attributes, fieldErrors field.ErrorList) *admissi
 
 // badRequestResponse is the response used when a request cannot be processed.
 func badRequestResponse(msg string) *admissionv1.AdmissionResponse {
-	return failureResponse(msg, metav1.StatusReasonBadRequest, http.StatusBadRequest)
+	response := failureResponse(msg, metav1.StatusReasonBadRequest, http.StatusBadRequest)
+	response.AuditAnnotations = map[string]string{"error": msg}
+	return response
 }
 
 // internalErrorResponse is the response used for unexpected errors
 func internalErrorResponse(msg string) *admissionv1.AdmissionResponse {
-	return failureResponse(msg, metav1.StatusReasonInternalError, http.StatusInternalServerError)
+	response := failureResponse(msg, metav1.StatusReasonInternalError, http.StatusInternalServerError)
+	response.AuditAnnotations = map[string]string{"error": msg}
+	return response
 }
 
 // isSignificantPodUpdate determines whether a pod update should trigger a policy evaluation.
 // Relevant mutable pod fields as of 1.21 are image and seccomp annotations:
 // * https://github.com/kubernetes/kubernetes/blob/release-1.21/pkg/apis/core/validation/validation.go#L3947-L3949
 func isSignificantPodUpdate(pod, oldPod *corev1.Pod) bool {
+	// TODO: invert this logic to allow specific update types.
```

Comment:
I'm not sure I understand the TODO

Reply:
Currently we're enumerating the cases that call for re-evaluating the policy. I'd rather enumerate the cases that we can safely ignore, incase we ever relax the mutation restrictions in the future (or just missed something...). It makes the logic more complicated here, but means a safer failure scenario.

Reply:
ah, ok